plugins/de.cau.cs.kieler.klighd.lsp/src/de/cau/cs/kieler/klighd/lsp/gson_utils/KGraphTypeAdapterUtil.xtend

```diff
@@ -37,42 +37,80 @@ import java.awt.geom.Point2D
+import de.cau.cs.kieler.klighd.lsp.structuredProgramming.sccharts.ChangeTargetStateAction
+import de.cau.cs.kieler.klighd.lsp.structuredProgramming.sccharts.ChangeSourceStateAction
+import de.cau.cs.kieler.klighd.lsp.structuredProgramming.sccharts.ChangeToAbortingTransitionAction
+import de.cau.cs.kieler.klighd.lsp.structuredProgramming.sccharts.ChangeToTerminatingTransitionAction
+import de.cau.cs.kieler.klighd.lsp.structuredProgramming.sccharts.ChangeToWeakTransitionAction
+import de.cau.cs.kieler.klighd.lsp.structuredProgramming.sccharts.AddTransitionAction
+import de.cau.cs.kieler.klighd.lsp.structuredProgramming.sccharts.ToggleFinalStateAction
+import de.cau.cs.kieler.klighd.lsp.structuredProgramming.sccharts.EditSemanticDeclarationAction
+import de.cau.cs.kieler.klighd.lsp.structuredProgramming.sccharts.MakeInitialStateAction
+import de.cau.cs.kieler.klighd.lsp.structuredProgramming.sccharts.ChangePriorityAction
+
 /**
  * Static util class to configure needed gson type adapters for KGraph serialization.
  * 
  * @author nre
  */
 class KGraphTypeAdapterUtil {
     def static GsonBuilder configureGson(GsonBuilder gsonBuilder) {
-        gsonBuilder
-        .registerTypeAdapterFactory(
+        gsonBuilder.registerTypeAdapterFactory(
             new ActionTypeAdapter.Factory => [
                 // General Sprotty actions
                 addActionKind(CheckedImagesAction.KIND, CheckedImagesAction)
                 addActionKind(PerformActionAction.KIND, PerformActionAction)
                 addActionKind(SetSynthesisAction.KIND, SetSynthesisAction)
                 addActionKind(RefreshDiagramAction.KIND, RefreshDiagramAction)
                 addActionKind(RefreshLayoutAction.KIND, RefreshLayoutAction)
-                
+
                 // Interactive layered actions
                 addActionKind(SetStaticConstraintAction.KIND, SetStaticConstraintAction)
                 addActionKind(SetPositionConstraintAction.KIND, SetPositionConstraintAction)
                 addActionKind(SetLayerConstraintAction.KIND, SetLayerConstraintAction)
                 addActionKind(DeleteStaticConstraintAction.KIND, DeleteStaticConstraintAction)
                 addActionKind(DeletePositionConstraintAction.KIND, DeletePositionConstraintAction)
                 addActionKind(DeleteLayerConstraintAction.KIND, DeleteLayerConstraintAction)
-                
+
                 // Interactive rectpacking actions
                 addActionKind(RectpackingSetPositionConstraintAction.KIND, RectpackingSetPositionConstraintAction)
                 addActionKind(RectpackingDeletePositionConstraintAction.KIND, RectpackingDeletePositionConstraintAction)
                 addActionKind(SetAspectRatioAction.KIND, SetAspectRatioAction)
-                
+
                 // Incremental topdown actions
                 addActionKind(RequestDiagramPieceAction.KIND, RequestDiagramPieceAction)
+
+                // Structured Programming actions
```

Comment:
Maybe add a service interface for a type adapter that allows adding actions for different languages.